lib/Client.php

```diff
@@ -181,6 +181,13 @@ protected function fetchURL($url, $post_body = null, $headers = []) {
 		return parent::fetchURL($url, $post_body, $headers);
 	}
 
+	/**
+	 * @codeCoverageIgnore
+	 */
+	public function getCodeChallengeMethod() {
+		return 'S256';
```

Comment:
has to be a config parameter - this is not true for all IdPs on this planet

Reply:
@DeepDiver1975 This is as far as I understood to signify which challenge methods the Client supports, the IDP capability is always read from the .well-known. 

See: https://github.com/jumbojett/OpenID-Connect-PHP/blob/master/src/OpenIDConnectClient.php#L664

Reply:
Also See: https://github.com/jumbojett/OpenID-Connect-PHP/blob/master/src/OpenIDConnectClient.php#L1812

I supose it is built like that to accomodate for implementations which extend from the jumbojett base client.

Reply:
You are right! I should have read the code base more in depth!